### ISSUE TITLE
./letsencrypt-auto: line 1810: tall: command not found

### ISSUE THREAD

**Issue by hach-que** [letsencrypt/letsencrypt#2572]
I'm getting this when trying to run ./letsencrypt-auto.  It says it's upgrading from 0.4.0 to 0.4.1 immediately after cloning from the Git repository:

```
Bootstrapping dependencies for openSUSE-based OSes...

The following 26 NEW packages are going to be installed:
  augeas-lenses binutils cpp cpp48 dialog gcc gcc48 glibc-devel libasan0 libatomic1 libcloog-isl4 libdialog11 libffi48-devel libgomp1 libisl10 libitm1 libmpc3 libmpfr4 libopenssl-devel libtsan0 linux-glibc-devel pkg-config python-devel python-pip python-virtualenv zlib-devel 

The following package is suggested, but will not be installed:
  terminfo 

26 new packages to install.
Overall download size: 23.6 MiB. After the operation, additional 104.8 MiB will be used.
Continue? [y/n/? shows all options] (y): y
Checking for new version...
Upgrading letsencrypt-auto 0.4.0 to 0.4.1...
Replacing letsencrypt-auto...
   cp /tmp/tmp.LhNZGKOTqT/letsencrypt-auto ./letsencrypt-auto
Bootstrapping dependencies for openSUSE-based OSes...

Creating virtual environment...
Installing Python packages...
Installation succeeded.
Requesting root privileges to run letsencrypt...
   /root/.local/share/letsencrypt/bin/letsencrypt --no-self-upgrade --help

  letsencrypt-auto [SUBCOMMAND] [options] [-d domain] [-d domain] ...

The Let's Encrypt agent can obtain and install HTTPS/TLS/SSL certificates.  By
default, it will attempt to use a webserver both for obtaining and installing
the cert. Major SUBCOMMANDS are:

  (default) run        Obtain & install a cert in your current webserver
  certonly             Obtain cert, but do not install it (aka "auth")
  install              Install a previously obtained cert in a server
  renew                Renew previously obtained certs that are near expiry
  revoke               Revoke a previously obtained certificate
  rollback             Rollback server configuration changes made during install
  config_changes       Show changes made to server config during installation
  plugins              Display information about installed plugins

Choice of server plugins for obtaining and installing cert:

  --apache          Use the Apache plugin for authentication & installation
  --standalone      Run a standalone webserver for authentication
  (nginx support is experimental, buggy, and not installed by default)
  --webroot         Place files in a server's webroot folder for authentication

OR use different plugins to obtain (authenticate) the cert and then install it:

  --authenticator standalone --installer apache

More detailed help:

  -h, --help [topic]    print this message, or detailed help on a topic;
                        the available topics are:

   all, automation, paths, security, testing, or any of the subcommands or
   plugins (certonly, install, nginx, apache, standalone, webroot, etc)

./letsencrypt-auto: line 1810: tall: command not found
The command '/bin/sh -c ./letsencrypt-auto --help' returned a non-zero code: 127
```

This doesn't make sense to me; it doesn't look like letsencrypt-auto even has a line 1810, and there's no reference to a "tall" command on the internet?

**Comment by hach-que** [comment]
I just noticed that these two files are different in the repository?

```
/letsencrypt-auto
/letsencrypt-auto-source/letsencrypt-auto
```

One has been changed 4 days ago and one has been changed 11 days ago?

**Comment by hach-que** [comment]
One of them reports itself as 0.4.0 and the other reports itself as 0.5.0.dev0, so I have no idea what the contents of 0.4.1 is or whether I should be trusting it given that the source doesn't seem to be available anywhere???

Running `./letsencrypt-auto-source/letsencrypt-auto --help` to do the initial self-upgrade actually works though, as it "upgrades" from `0.5.0.dev0` to `0.4.1`, but doesn't have the command not found error.

**Comment by bmw** [comment]
> This doesn't make sense to me; it doesn't look like letsencrypt-auto even has a line 1810, and there's no reference to a "tall" command on the internet?

I suspect the cause of this is #2456 (tl;dr: the script overwrites itself when updating which can cause problems in some shells). This was fixed in the 0.4.1 release of `letsencrypt-auto`, however, older versions of the script trying to update themselves to 0.4.1 may still have this problem.

> I just noticed that these two files are different in the repository?

Yes. So the file `/letsencrypt-auto-source/letsencrypt-auto` is the latest, most bleeding edge version of the script. We don't expect/want most users to be running this and `/letsencrypt-auto` is the version of the script from the latest release. Because #2569 hasn't landed yet, the version here is still 0.4.0.

> One of them reports itself as 0.4.0 and the other reports itself as 0.5.0.dev0, so I have no idea what the contents of 0.4.1 is or whether I should be trusting it given that the source doesn't seem to be available anywhere???

#2569 should land soon, after which `/letsencrypt-auto` will be version 0.4.1 and using this version of the script will work for you. In the meantime, you can use the script available [here](https://github.com/letsencrypt/letsencrypt/blob/v0.4.1/letsencrypt-auto) which is what `letsencrypt-auto` is downloading when it upgrades to 0.4.1.

Sorry for any confusion and I hope this answers your questions.

**Comment by erikrose** [comment]
#2547 has landed in 0.4.1, and 0.4.1 is released. Please reopen if this reoccurs. Thanks for the report!
